src/fu-util-common.c

```diff
@@ -1490,9 +1490,8 @@ fu_util_plugin_flag_to_string(FwupdPluginFlags plugin_flag)
 		return NULL;
 	if (plugin_flag == FWUPD_PLUGIN_FLAG_USER_WARNING)
 		return NULL;
-	if (plugin_flag == FWUPD_PLUGIN_FLAG_REQUIRE_HWID)
-		return NULL;
-	if (plugin_flag == FWUPD_PLUGIN_FLAG_NONE) {
+	if (plugin_flag == FWUPD_PLUGIN_FLAG_NONE ||
+	    plugin_flag == FWUPD_PLUGIN_FLAG_REQUIRE_HWID) {
```

Comment:
but that's not right, surely:

		g_debug("enabling %s due to HwId %s", plugins[i], hwid);
		fu_plugin_remove_flag(plugin, FWUPD_PLUGIN_FLAG_REQUIRE_HWID);

i.e. the engine removes the REQUIRE_HWID flag if the HWID matches, otherwise the plugin remains disabled.

Reply:
Then - maybe the right answer is to let the engine run and then show enabled plugins.

Reply:
> let the engine run

I'm pretty sure that won't be useful; that'll take a few *seconds* and we want results in ms for user interactions...